elasticsearch/src/test/java/org/hibernate/search/backend/elasticsearch/test/ElasticsearchDSLIT.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * Hibernate Search, full-text search for your domain model
+ *
+ * License: GNU Lesser General Public License (LGPL), version 2.1 or later
+ * See the lgpl.txt file in the root directory or <http://www.gnu.org/licenses/lgpl-2.1.html>.
+ */
+package org.hibernate.search.backend.elasticsearch.test;
+
+import static org.fest.assertions.Assertions.assertThat;
+
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.Id;
+
+import org.apache.lucene.search.Query;
+import org.hibernate.Session;
+import org.hibernate.search.FullTextQuery;
+import org.hibernate.search.FullTextSession;
+import org.hibernate.search.Search;
+import org.hibernate.search.annotations.Analyzer;
+import org.hibernate.search.annotations.Field;
+import org.hibernate.search.annotations.Indexed;
+import org.hibernate.search.query.dsl.QueryBuilder;
+import org.hibernate.search.test.SearchTestBase;
+import org.junit.Test;
+
+/**
+ * Test the Query DSL with Elasticsearch
+ *
+ * @author Davide D'Alto
+ */
+public class ElasticsearchDSLIT extends SearchTestBase {
+
+	@Test
+	public void testDSLMatchAll() throws Exception {
+		try ( Session session = openSession() ) {
+			FullTextSession fullTextSession = Search.getFullTextSession( session );
+			final QueryBuilder queryBuilder = queryBuilder( fullTextSession );
+
+			Query query = queryBuilder
+					.all()
+					.createQuery();
+
+			FullTextQuery fullTextQuery = fullTextSession.createFullTextQuery( query, Letter.class );
+			String queryString = fullTextQuery.getQueryString();
+			assertThat( queryString ).isEqualTo( "{\"query\":{\"match_all\":{}}}" );
+		}
+	}
+
+	@Test
+	public void testDSLPhrase() throws Exception {
+		try ( Session session = openSession() ) {
+			FullTextSession fullTextSession = Search.getFullTextSession( session );
+			final QueryBuilder queryBuilder = queryBuilder( fullTextSession );
+
+			Query query = queryBuilder
+					.phrase()
+						.onField( "message" )
+						.sentence( "A very important matter" )
+					.createQuery();
+
+			FullTextQuery fullTextQuery = fullTextSession.createFullTextQuery( query, Letter.class );
+			String queryString = fullTextQuery.getQueryString();
+			assertThat( queryString ).isEqualTo( "{\"query\":{\"match_phrase\":{\"message\":{\"query\":\"very important matter\",\"slop\":0,\"boost\":1.0}}}}" );
```

Comment:
No need to change right now, but it's still good to know: We have `JsonHelper#assertJsonEquals()`  which allows you to use _'_ instead of the escaped _"_, making it a tad nicer to read.